backend/src/apiserver/storage/task_store.go

```diff
@@ -0,0 +1,173 @@
+	"github.com/golang/glog"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/common"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/list"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/model"
+	"github.com/kubeflow/pipelines/backend/src/common/util"
+)
+
+const table_name = "tasks"
+
+var (
+	taskColumns = []string{
+		"UUID",
+		"Namespace",
+		"PipelineName",
+		"MLMDExecutionID",
+		"CreatedTimestamp",
+		"StorageState",
+		"EndedTimestamp",
+		"Fingerprint",
+	}
+)
+
+type TaskStoreInterface interface {
+	// Create a task entry in the database
+	CreateTask(task *model.Task) (*model.Task, error)
+
+	ListTasks(filterContext *common.FilterContext, opts *list.Options) ([]*model.Task, int, string, error)
+}
+
+type TaskStore struct {
+	db   *DB
+	time util.TimeInterface
+	uuid util.UUIDGeneratorInterface
+}
+
+func (s *TaskStore) CreateTask(task *model.Task) (*model.Task, error) {
+	// Set up UUID for task.
+	newTask := *task
+	id, err := s.uuid.NewRandom()
+	if err != nil {
+		return nil, util.NewInternalServerError(err, "Failed to create an task id.")
+	}
+	newTask.UUID = id.String()
+
+	sql, args, err := sq.
+		Insert(table_name).
+		SetMap(sq.Eq{
+			"UUID":             newTask.UUID,
+			"Namespace":        newTask.Namespace,
+			"PipelineName":     newTask.PipelineName,
+			"RunUUID":          newTask.RunUUID,
+			"MLMDExecutionID":  newTask.MLMDExecutionID,
+			"CreatedTimestamp": newTask.CreatedTimestamp,
+			"EndedTimestamp":   newTask.EndedTimestamp,
+			"Fingerprint":      newTask.Fingerprint,
+		}).
+		ToSql()
+	if err != nil {
+		return nil, util.NewInternalServerError(err, "Failed to create query to insert task to task table: %v",
+			err.Error())
+	}
+	_, err = s.db.Exec(sql, args...)
+	if err != nil {
+		return nil, util.NewInternalServerError(err, "Failed to add task to task table: %v",
+			err.Error())
+	}
+	return &newTask, nil
+}
+
+func (s *TaskStore) scanRows(rows *sql.Rows) ([]*model.Task, error) {
+	var tasks []*model.Task
+	for rows.Next() {
+		var uuid, namespace, pipelineName, runUUID, mlmdExecutionID, fingerprint string
+		var createdTimestamp, endedTimestamp int64
+		err := rows.Scan(&uuid, &namespace, &pipelineName, &runUUID, &namespace, &mlmdExecutionID, createdTimestamp, endedTimestamp, fingerprint)
+		if err != nil {
+			return tasks, err
+		}
+		task := &model.Task{
+			UUID:             uuid,
+			Namespace:        namespace,
+			PipelineName:     pipelineName,
+			RunUUID:          runUUID,
+			MLMDExecutionID:  mlmdExecutionID,
+			CreatedTimestamp: createdTimestamp,
+			EndedTimestamp:   endedTimestamp,
+			Fingerprint:      fingerprint,
+		}
+		tasks = append(tasks, task)
+	}
+	return tasks, nil
+}
+
+// Runs two SQL queries in a transaction to return a list of matching experiments, as well as their
+// total_size. The total_size does not reflect the page size.
+func (s *TaskStore) ListTasks(filterContext *common.FilterContext, opts *list.Options) ([]*model.Task, int, string, error) {
```

Comment:
for discussion only: boilerplate code for implementing these queries are non-trivial. if you know any better tools to do this, feel free to bring them up later.